mono/tests/wait-dup-all.cs

```diff
@@ -0,0 +1,26 @@
+/*
+https://github.com/mono/mono/issues/9089
+Duplication is ok for WaitAny, exception for WaitAll.
+System.DuplicateWaitObjectException: Duplicate objects in argument.
+*/
+using System;
+using System.Threading;
+
+class Program
```

Comment:
Those tests would be better included in the corlib test suite. mono/tests is better for tests that must be standalone or require custom process setup.

Reply:
This area of the tree is much easier for me to work with, which I realize is not the best reason. Editing the makefile, editing-existing/adding-new/running tests, all much easier here.